### PR TITLE
Serialport 5 upgrade and Bugflix 2.1.0 release.

### PR DESCRIPTION
**Pull request by techninja**
Ok! This PR is meant to comprise all minimum work required for the 2.1.0 bugfix release that was due months ago but has been offset by external forces.

The testing of this branch should allow the release to be made fully real, and this text will comprise something of a beta release notes for 2.1.0 when testing is done.

## Updated/Added:
- Electron from 1.0.2 to 1.6.10 (Lots of breaking change updates go along with this)
  - Allows for per mode reloads with preload scripts, among other innumerable bugfixes and speed improvements.
- Root jQuery 2.2.3 to 3.2.1, paper.js 0.9.25 to 0.11.3, qTip2 2.2.0 to 3.03 
  - Includes jQuery migrate legacy plugin for older code (in edit mode, etc).
- [Devtron: for advanced debugging!](https://electron.atom.io/devtron/)
- Winston: For advanced logging! See readme for more info and log file locations.
- Pen defaults for machines with assumed pen only now work better, and "auto display size" works as expected with this.
- Dev `npm run rebuild` added for serialport build upgrades in the future.
- Global startup error catch expanded, and now you can copy the error on screen for reference (couldn't do that at all before!).

## Bugfixes:
- Printing a lot from windows was basically impossible without errors, but now works wonderfully
- Disconnect detection was broken for a few versions, but should now be working again.
- Fixed a bunch of legacy translation and settings code to better conform with cross-os path standards.

## Known issues:
- Logging is somewhat hacky and does not capture _everything_ that's logged, just most of it for known  (and a few unknown) logging situations. Should work for now, but need to ensure that common error prone code logs more of its important data or errors.

### CODE REVIEW

**Review by docprofsky — COMMENTED** [review]
Other than the spelling, this seems good. Note that I have only skimmed this and not yet tested on hardware.

**Review by rogerfachini — APPROVED** [review]
Release tested on hardware, no major issues found. Code looks good, pinging @docprofsky for a more experienced review of the code.

File: README.md [hunk 1]
```diff
@@ -72,6 +70,16 @@ instead of this one.
 is covering your question or problem, if not, ask away! Someone will be around
 to help soon.
 
+### Logging
+RoboPaint 2.1.0 introduces full application logging. If you have a problem
+starting or running the app, submit your logs with your issue or to Evil Mad
+Scientist and we should be able to help you a lot more directly.
+
+**Log data locations:**
```
[the remark this answers is not in the document]
Comment (techninja) [reply README.md]: Good idea, will add that here.

[the remark this answers is not in the document]
Comment (techninja) [reply README.md]: On second thought, the text above does everything needed here, I'll just link to this from the issue template.

**Pull request merged**